File: Emulator/src/main/content/global/skill/summoning/familiar/npc/LavaTitanOptionPlugin.java
package content.global.skill.summoning.familiar.npc;

import core.cache.def.impl.NPCDefinition;
import core.game.interaction.OptionHandler;
import core.game.node.Node;
import core.game.node.entity.player.Player;
import core.plugin.Initializable;
import core.plugin.Plugin;

/**
 * The type Lava titan option plugin.
 */
@Initializable
public final class LavaTitanOptionPlugin extends OptionHandler {

    @Override
    public Plugin<Object> newInstance(Object arg) throws Throwable {
        NPCDefinition.forId(7341).getHandlers().put("option:interact", this);
        return this;
    }

    @Override
    public boolean handle(final Player player, Node node, String option) {
        player.getDialogueInterpreter().open(8700, node.asNpc());
        return true;
    }

}

File: Emulator/src/main/content/global/skill/summoning/familiar/npc/LavaTitanOptionPlugin.kt
package content.global.skill.summoning.familiar.npc

import core.cache.def.impl.NPCDefinition
import core.game.interaction.OptionHandler
import core.game.node.Node
import core.game.node.entity.player.Player
import core.plugin.Initializable
import core.plugin.Plugin
import shared.consts.NPCs

@Initializable
class LavaTitanOptionPlugin : OptionHandler() {
    @Throws(Throwable::class)
    override fun newInstance(arg: Any?): Plugin<Any> {
        NPCDefinition.forId(NPCs.LAVA_TITAN_7341).handlers["option:interact"] = this
        return this
    }

    override fun handle(player: Player, node: Node, option: String): Boolean {
        player.dialogueInterpreter.open(8700, node.asNpc())
        return true
    }
}
